applications/FluidDynamicsApplication/custom_constitutive/newtonian_3d_law.cpp

```diff
@@ -99,13 +103,20 @@ std::string Newtonian3DLaw::Info() const {
     return "Newtonian3DLaw";
 }
 
-double Newtonian3DLaw::GetEffectiveViscosity(ConstitutiveLaw::Parameters& rParameters) const {
-    // We are abusing the fact that C(5,5) = mu
-    return rParameters.GetConstitutiveMatrix()(5,5);
-}
+double Newtonian3DLaw::GetEffectiveViscosity(ConstitutiveLaw::Parameters& rParameters) const
+{
+    double effective_viscosity;
+    const Properties &r_prop = rParameters.GetMaterialProperties();
+
+    if (r_prop.HasTable(TEMPERATURE, DYNAMIC_VISCOSITY)) {
```

Comment:
wont this have a performance effect? => this was one of the concerns when discussing #2349

Reply:
Probably yes... it's an if statement per Gauss pt. However we need this functionality to be working at the end of month because an industrial project, so we can't wait for #2349 to be merged...

Reply:
hm not sure if this is a valid reason, you could have it in a local branch...
Either way, not blocking

just saying that having #2349 would be nice to have. Lets see when I have time to finish it ... :)

Reply:
Totally agree... As soon as you finish #2349 we can upgrade this.